From f0991a389e1b263a68f2c730dba4187a8d1eefd4 Mon Sep 17 00:00:00 2001
From: Manikanta Kanamarlapudi <kmanikan@codeaurora.org>
Date: Fri, 7 Jul 2017 16:24:31 +0530
Subject: [PATCH 12/20] NdkMediaCodec: add AMediaCodec_getBufferFormat

Add new AMediaCodec_getBufferFormat api to get
format of the buffer of specified index.

Change-Id: Ib5d6537b601dc8ce9c5c0096485637e0cf44a16a
---
 media/ndk/NdkMediaCodec.cpp             | 7 +++++++
 media/ndk/include/media/NdkMediaCodec.h | 6 ++++++
 2 files changed, 13 insertions(+)

diff --git a/media/ndk/NdkMediaCodec.cpp b/media/ndk/NdkMediaCodec.cpp
index 128edba1c..11dedbb73 100644
--- a/media/ndk/NdkMediaCodec.cpp
+++ b/media/ndk/NdkMediaCodec.cpp
@@ -366,6 +366,13 @@ AMediaFormat* AMediaCodec_getOutputFormat(AMediaCodec *mData) {
     return AMediaFormat_fromMsg(&format);
 }
 
+EXPORT
+AMediaFormat* AMediaCodec_getBufferFormat(AMediaCodec *mData, size_t index) {
+    sp<AMessage> format;
+    mData->mCodec->getOutputFormat(index, &format);
+    return AMediaFormat_fromMsg(&format);
+}
+
 EXPORT
 media_status_t AMediaCodec_releaseOutputBuffer(AMediaCodec *mData, size_t idx, bool render) {
     if (render) {
diff --git a/media/ndk/include/media/NdkMediaCodec.h b/media/ndk/include/media/NdkMediaCodec.h
index 7e7e81ed0..56f8b395e 100644
--- a/media/ndk/include/media/NdkMediaCodec.h
+++ b/media/ndk/include/media/NdkMediaCodec.h
@@ -178,6 +178,12 @@ ssize_t AMediaCodec_dequeueOutputBuffer(AMediaCodec*, AMediaCodecBufferInfo *inf
         int64_t timeoutUs);
 AMediaFormat* AMediaCodec_getOutputFormat(AMediaCodec*);
 
+/**
+ * Get format of the buffer. The specified buffer index must have been previously obtained from
+ * dequeueOutputBuffer.
+ */
+AMediaFormat* AMediaCodec_getBufferFormat(AMediaCodec*, size_t index);
+
 /**
  * If you are done with a buffer, use this call to return the buffer to
  * the codec. If you previously specified a surface when configuring this
-- 
2.17.1

